cmake/std/atdm/doom/environment.sh

```diff
@@ -0,0 +1,292 @@
+
+# NOTE: The above does not work.  For some reason, the library 'iomp5' can't
+# be found at runtime.  Instead, you have to explicitly list out the library
+# files in order as shown below.  Very sad.
+
+atdm_config_add_libs_to_var ATDM_CONFIG_BLAS_LIBS ${ATDM_CONFIG_MKL_ROOT}/mkl/lib/intel64 .so \
+  mkl_intel_lp64 mkl_intel_thread mkl_core
+
+atdm_config_add_libs_to_var ATDM_CONFIG_BLAS_LIBS ${ATDM_CONFIG_MKL_ROOT}/lib/intel64 .so \
+  iomp5
+
+export ATDM_CONFIG_LAPACK_LIBS=${ATDM_CONFIG_BLAS_LIBS}
+
+# Boost
+
+atdm_config_add_libs_to_var ATDM_CONFIG_BOOST_LIBS ${BOOST_ROOT}/lib .a \
+  boost_program_options boost_system
+
+# NOTE: Above, the SPARC-installed TPLs only have *.a files.  There are no
+# *.so files.
+
+# HDF5 and Netcdf
+
+# NOTE: HDF5_ROOT and NETCDF_ROOT should already be set in env from above
+# module loads!
+
+# However, set the direct libs for HDF5 and NetCDF in case we use that option
+# for building (see env var ATDM_CONFIG_USE_SPARC_TPL_FIND_SETTINGS).
+
+export ATDM_CONFIG_HDF5_LIBS="-L${HDF5_ROOT}/lib;${HDF5_ROOT}/lib/libhdf5_hl.a;${HDF5_ROOT}/lib/libhdf5.a;-lz;-ldl"
+
+export ATDM_CONFIG_NETCDF_LIBS="-L${BOOST_ROOT}/lib;-L${NETCDF_ROOT}/lib;-L${NETCDF_ROOT}/lib;-L${SEMS_PNETCDF_ROOT}/lib;-L${HDF5_ROOT}/lib;${BOOST_ROOT}/lib/libboost_program_options.a;${BOOST_ROOT}/lib/libboost_system.a;${NETCDF_ROOT}/lib/libnetcdf.a;${PNETCDF_ROOT}/lib/libpnetcdf.a;${HDF5_ROOT}/lib/libhdf5_hl.a;${HDF5_ROOT}/lib/libhdf5.a;-lz;-ldl;-lcurl"
+
+# SuperLUDist
+if [[ "${ATDM_CONFIG_SUPERLUDIST_INCLUDE_DIRS}" == "" ]] ; then
+  # Set the default which is correct for all of the new TPL builds
+  export ATDM_CONFIG_SUPERLUDIST_INCLUDE_DIRS=${SUPERLUDIST_ROOT}/include
+  export ATDM_CONFIG_SUPERLUDIST_LIBS=${SUPERLUDIST_ROOT}/lib/libsuperlu_dist.a
+fi
+
+# Finished!
+export ATDM_CONFIG_COMPLETED_ENV_SETUP=TRUE
+
+#
+# Run a script on the compute node send STDOUT and STDERR output to a file
+# while also echo output to the console.  The primary purpose is to run the
+# tests on the compute node.
+#
+# Usage:
+#
+#   atdm_run_script_on_comput_node <script_to_run> <output_file> \
+#     [<timeout>] [<account>]
+#
+# If <timeout> and/or <account> are not given, then defaults are provided that
+# work for the Jenkins driver process.
+#
+# In this case, sbatch is used to run the script but it also sends ouptut to
+# STDOUT in real-time while it is running in addition to writing to the
+# <outout_file>.  The job name for the sbatch script is taken from the env var
+# 'ATDM_CONFIG_BUILD_NAME'.  This works for local builds since ATDM_CONFIG_BUILD_NAME.
+#
+# Note that you can pass in the script to run with arguments such as with
+# "<some-script> <arg1> <arg2>" and it will work.  But note that this has to
+# be bash script that 'sbatch' can copy and run form a temp location and it
+# still has to work.  So the script has to use absolute directory paths, not
+# relative paths or asume sym links, etc.
+#
+function atdm_run_script_on_compute_node {
```

Comment:
It would be good to find a way to not duplicate this function.  Could we put this in its own file and then reuse this across all system envs that use it?